src/CoreWCF.Queue/src/CoreWCF/Queue/Common/DefaultQueueTransportPump.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CoreWCF.Queue.Common
+{
+    /// <summary>
+    /// DefaultQueueTransportPump model serves the purpose of pull model
+    /// </summary>
+    public class DefaultQueueTransportPump : QueueTransportPump
+    {
```

Comment:
I think this can be internal as we have a factory method on QueueTransportPump to create an instance.